src/components/Profile/ProfileContainer.js

```diff
@@ -18,15 +17,13 @@ class ProfileContainer extends Component {
     if (!userId) {
       userId = DEFAULT_USER_ID;
     }
-
-    axios.get(`${MAIN_URL}profile/${userId}`)
-      .then((response) => {
-        props.setUserProfile(response.data);
-      });
+    props.getUserProfile(userId);
   }
 
   render() {
-    const { profile } = this.props;
+    const { profile, isAuth } = this.props;
+
+    if (isAuth === false) return <Redirect to="/login" />;
```

Comment:
Consider NOT comparing like this `smth === true/false`. This just implies that you don't really understand the JS types.
For example, to check if "isAuth" is falsy, you could just write `if (!isAuth) {...}`

Reply:
thx, i really should think about it
/fxd